File: DSAcpp/Algorithm/Searching/binarySearch.cpp
#include <iostream>
#include <vector>
using namespace std;

bool binSearch(const vector<int> &arr,const int x){
    int lw = 0, hi = arr.size()-1,mid;

    while (lw<=hi)
    {
        mid = (hi+lw)>>1;
        if(arr[mid]==x) return true;
        else if(arr[mid]<x) lw = mid+1;
        else hi = mid-1;
    }
    return false;
}

int main(){
    vector <int> arr(20);
    for(int i=0;i<20;i++) arr[i] = (i+1)*3;

    for(int i=1;i<=40;i++){
        cout << i << "\t: " << (binSearch(arr,i)?"YES\n":"NO\n");
    }
}

File: DSApython/Algorithm/Searching/binarySearch.py

def binarySearch(arr,a):
    lw,hi = 0,len(arr)-1

    while lw<=hi:
        mid = (hi+lw)>>1
        if arr[mid] == a: return True
        elif arr[mid] < a: lw = mid+1
        else : hi = mid-1
    return False

arr = [(i+1)*2 for i in range(20)]

for i in range(1,40+1):
    print(f"{i}\t: {"Yes" if binarySearch(arr,i) else "NO"}")
